src/Root.tsx

```diff
@@ -48,6 +49,27 @@ export async function onLatestVersionChange(newVer: SemVer | null | undefined, o
 const Root = observer(() => {
   const { app, modals } = getRootStore();
 
+  React.useEffect(() => {
+    const searchParams = new URLSearchParams(document.location.search);
+    const result = searchParams.get("Diagram");
+    if (result === null) return;
+
+    // Replace URL-safe characters with base64 equivalents
+    let base64String = result.replace(/-/g, "+").replace(/_/g, "/");
+
+    // Add padding if necessary
+    while (base64String.length % 4 !== 0) {
+      base64String += "=";
+    }
+    console.log(base64String.length);
+
+    const jsonDiagram = window.atob(base64String);
+    app.diagram = plainToClass(Diagram, JSON.parse(jsonDiagram), {
```

Comment:
Add try-catch

Reply:
Try-catch added, will continue for implement modal panel for handle the error.